engine/src/main/java/org/terasology/engine/core/module/ModuleManager.java

```diff
@@ -73,17 +73,32 @@ public ModuleManager(String masterServerAddress, List<Class<?>> classesOnClasspa
 
         registry = new TableModuleRegistry();
 
-        engineModule = loadEngineModule(classesOnClasspathsToAddToEngine);
+        engineModule = moduleFactory.createPackageModule("org.terasology.engine");
+        Module nui = createSyntheticPackageModule("nui", "3.0.0", "org.terasology.nui");
+        Module discordSubsystem = moduleFactory.createPackageModule("org.terasology.subsystem.discordrpc");
 
-        loadModulesFromApplicationPath(pathManager);
+        createDependency(engineModule, nui);
+
+        registry.add(nui);
+        registry.add(engineModule);
+        registry.add(discordSubsystem);
 
+        loadModulesFromApplicationPath(pathManager);
         ensureModulesDependOnEngine();
 
         setupSandbox();
-        loadEnvironment(Sets.newHashSet(engineModule), true);
+        loadEnvironment(Sets.newHashSet(engineModule, nui, discordSubsystem), true);
         installManager = new ModuleInstallManager(this, masterServerAddress);
     }
 
+    private  void createDependency(Module parent, Module child) {
+        DependencyInfo dependencyInfo = new DependencyInfo();
+        dependencyInfo.setId(child.getId());
+        dependencyInfo.setMinVersion(child.getVersion());
+        dependencyInfo.setMaxVersion(child.getVersion().getNextPatchVersion());
+        parent.getMetadata().getDependencies().add(dependencyInfo);
```

Comment:
👍 makes sense to have a function for this.

TODO: refactor `ensureModulesDependOnEngine` method (just below this one) to use this.

FIXME: note https://github.com/MovingBlocks/Terasology/pull/4593/commits/20546f1853a8f5cce99fff8018700d81340181d3 — getNextPatchVersion may break for this purpose when base version is a SNAPSHOT.